app/controllers/submissions_controller.rb

```diff
@@ -509,39 +507,42 @@ def set_testrun
     }
   end
 
-  def transmit_lti_score
-    responses = send_scores(@submission)
-    failed_users = []
-    too_late = false
-    status = []
-    score_sent = nil
-
-    responses.each do |response|
-      if Lti::ERROR_STATUS.include? response[:status]
-        failed_users << response[:user]
-      elsif response[:score_sent] != @submission.normalized_score # the score was sent successfully, but received too late
-        too_late = true
-        score_sent = response[:score_sent]
-      end
+  def check_scoring_too_late(submit_info, client_socket)
```

Comment:
I would be open to have this message appear in all cases of the `transmit_lti_score` method, further reducing the complexity of the code (it is, indeed, independent of the LTI stuff, isn't it?).

Reply:
We talked about it that the notification is **not** shown when a failure occurred because then it is uninteresting if the score was reduced.